tests/unit/amo/server/test_base.js

```diff
@@ -560,6 +560,115 @@ describe(__filename, () => {
       expect(users.currentUserWasLoggedOut).toEqual(true);
       mockUsersApi.verify();
     });
+
+    it('sets a X-Accel-Expires header if request is safe & anonymous and response is sucessful', async () => {
+      const { store, sagaMiddleware } = createStoreAndSagas();
+
+      const response = await testClient({ store, sagaMiddleware }).get(
+        '/en-US/firefox/',
+      );
+      expect(response.headers['X-Accel-Expires'.toLowerCase()]).toEqual('180');
+    });
+
+    it('does not set a X-Accel-Expires header if request contained token cookie', async () => {
+      const { store, sagaMiddleware } = createStoreAndSagas();
+
+      const response = await testClient({ store, sagaMiddleware })
+        .get('/en-US/firefox/')
+        .set('cookie', `${defaultConfig.get('cookieName')}="foo"`);
+      expect(response.headers['X-Accel-Expires'.toLowerCase()]).toBeUndefined();
+    });
+
+    it('does not set a X-Accel-Expires header if request method is not read-only', async () => {
+      const { store, sagaMiddleware } = createStoreAndSagas();
+
+      const response = await testClient({ store, sagaMiddleware }).post(
+        '/en-US/firefox/',
+      );
+      expect(response.headers['X-Accel-Expires'.toLowerCase()]).toBeUndefined();
+    });
+
+    it('does not set a X-Accel-Expires header if response is 404', async () => {
+      // Component and App setup to generate a 404 stolen from a previous test
+      // above.
+      class NotFound extends React.Component {
+        render() {
+          return (
+            <NestedStatus code={404}>
+              <h1>Not Found</h1>
+            </NestedStatus>
+          );
+        }
+      }
+
+      const NotFoundApp = () => (
+        <div>
+          <Route path="*" component={NotFound} />
+        </div>
+      );
+
+      const response = await testClient({ App: NotFoundApp }).get(
+        '/en-US/firefox/simulation-of-a-non-existent-page/',
+      );
+      expect(response.statusCode).toEqual(404);
+      expect(response.headers['X-Accel-Expires'.toLowerCase()]).toBeUndefined();
+    });
+
+    it('does not set a X-Accel-Expires header if response is 500', async () => {
+      const _createHistory = () => {
+        throw new Error('oops');
+      };
+
+      const response = await testClient({ _createHistory }).get(
+        '/en-US/firefox/',
+      );
+      expect(response.statusCode).toEqual(500);
+      expect(response.headers['X-Accel-Expires'.toLowerCase()]).toBeUndefined();
```

Comment:
Maybe something like this would read better:

```suggestion
      expect(response.headers).not.toContain(X_ACCEL_HEADER);
```